ca/certificates.go

```diff
@@ -227,23 +228,25 @@ func (rca *RootCA) RequestAndSaveNewCertificates(ctx context.Context, kw KeyWrit
 		}
 	}
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	// If a CA certificate bundle exists it has been validated before. If it's different, let's write it to disk.
 	// Root rotation should always happen by appending a new CA cert, and later removing the old one,
 	// so it's safer to do it in this order of operations (write root, write certificate)
+	var returnRootCA *RootCA
 	if newRootErr == nil && !bytes.Equal(rca.Cert, response.RootCABundle) {
 		if err := newRootCA.saveCertificate(); err != nil {
-			return nil, err
+			return nil, nil, err
 		}
+		returnRootCA = &newRootCA
```

Comment:
`newRootCA` was configured to create certs that will only be valid for 1 minute, so it should not be returned for long-term use.